tests/unit/Storefront/Controller/AccountProfileControllerTest.php

```diff
@@ -0,0 +1,190 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Tests\Unit\Storefront\Controller;
+
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\TestCase;
+use Shopware\Core\Checkout\Customer\CustomerEntity;
+use Shopware\Core\Checkout\Customer\SalesChannel\ChangePasswordRoute;
+use Shopware\Core\Framework\Routing\RoutingException;
+use Shopware\Core\Framework\Validation\DataBag\RequestDataBag;
+use Shopware\Core\Framework\Validation\Exception\ConstraintViolationException;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+use Shopware\Storefront\Controller\AccountProfileController;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Validator\ConstraintViolationList;
+
+/**
+ * @internal
+ */
+#[CoversClass(AccountProfileController::class)]
+class AccountProfileControllerTest extends TestCase
+{
+    public function testSavePasswordWithMissingPasswordParam(): void
+    {
+        $controller = $this->createAccountProfileController();
+        $dataBag = new RequestDataBag();
+
+        $this->expectException(RoutingException::class);
+        $this->expectExceptionMessage('Parameter "password" is missing.');
+
+        $controller->savePassword(
+            $dataBag,
+            $this->createMock(SalesChannelContext::class),
+            new CustomerEntity(),
+            new Request()
+        );
+    }
+
+    public function testSavePasswordWithConstraintViolation(): void
+    {
+        $controller = $this->createAccountProfileController(true);
+
+        $passwordBag = new RequestDataBag([
+            'newPassword' => 'newPassword123',
+            'newPasswordConfirm' => 'newPassword123',
+            'password' => 'oldPassword',
+        ]);
+
+        $dataBag = new RequestDataBag(['password' => $passwordBag]);
+
+        $response = $controller->savePassword(
+            $dataBag,
+            $this->createMock(SalesChannelContext::class),
+            new CustomerEntity(),
+            new Request()
+        );
+
+        static::assertEquals('frontend.account.profile.page', $response->headers->get('X-Forwarded-Route'));
+    }
+
+    public function testSavePasswordWithDefaultRedirect(): void
+    {
+        $controller = $this->createAccountProfileController();
+
+        $passwordBag = new RequestDataBag([
+            'newPassword' => 'newPassword123',
+            'newPasswordConfirm' => 'newPassword123',
+            'password' => 'oldPassword',
+        ]);
+
+        $dataBag = new RequestDataBag(['password' => $passwordBag]);
+
+        $response = $controller->savePassword(
+            $dataBag,
+            $this->createMock(SalesChannelContext::class),
+            new CustomerEntity(),
+            new Request()
+        );
+
+        static::assertEquals('frontend.account.profile.page', $response->headers->get('X-Redirect-Route'));
```

Comment:
please use `assertSame` where possible instead, as it is type safe

this is the same assertion as the test before, so it is not really tested, that this is the successful case and the one above the failing case